### PR TITLE
Stabilization v0.1.47

### PR DESCRIPTION
**Pull request by yuumasato**
#### Description:

- Merge fixes done for v0.1.47 into master, so they are part of next release.

### CODE REVIEW

**Comment by yuumasato** [comment]
Ok, the approach of trying to merge via GH is much cleaner on what is being merged into master.

But I don't know how conflict resolution will work.

**Comment by ggbecker** [comment]
> Ok, the approach of trying to merge via GH is much cleaner on what is being merged into master.
> 
> But I don't know how conflict resolution will work.

One way to resolve the conflict is to rebase against master and resolve the conflict on `tests/test_parse_affected.py` which seemed to be the real conflict here. But then all the commits will have someone else's as author as you can see here: https://github.com/ComplianceAsCode/content/pull/4984

**Comment by yuumasato** [comment]
> One way to resolve the conflict is to rebase against master and resolve the conflict on `tests/test_parse_affected.py` which seemed to be the real conflict here.

It is easy to resolve the conflict on CLI, but then it will go directly to master, and there won't be review of the conflict resolution.

> But then all the commits will have someone else's as author as you can see here: #4984

This rewriting the history.

**Comment by ggbecker** [comment]
> > One way to resolve the conflict is to rebase against master and resolve the conflict on `tests/test_parse_affected.py` which seemed to be the real conflict here.
> 
> It is easy to resolve the conflict on CLI, but then it will go directly to master, and there won't be review of the conflict resolution.
> 
> > But then all the commits will have someone else's as author as you can see here: #4984
> 
> This rewriting the history.

I see, what about merging master to stabilization branch as suggested by github?
```
git fetch origin
git checkout -b stabilization-v0.1.47 origin/stabilization-v0.1.47
git merge master
```
Then it will contain one new specific commit in stabilization branch which contains the actual conflict resolution.

**Comment by jan-cerny** [comment]
Hi @yuumasato, do it normally on the command line. Make sure you pulled everything, then checkout master branch, merge in the stabilization branch, solve the conflicts and push to upstream. We do it this way in OpenSCAP and it always worked.

**Comment by yuumasato** [comment]
I just merged locally and pushed to master.
Food for thought: For next releases backporting the patches instead of forward porting them may be easier...

**Comment by yuumasato** [comment]
Há, interestingly GitHub is able to track the merge here.

**Comment by yuumasato** [comment]
> Hi @yuumasato, do it normally on the command line. Make sure you pulled everything, then checkout master branch, merge in the stabilization branch, solve the conflicts and push to upstream. We do it this way in OpenSCAP and it always worked.

@jan-cerny Sure, it works. I just would like to try if something makes review of conflict resolution possible.

File: tests/test_parse_affected.py [hunk 1]
```diff
@@ -45,7 +45,11 @@ def main():
 
         for rule_dir in ssg.rules.find_rule_dirs(guide_dir):
             rule_path = os.path.join(rule_dir, "rule.yml")
-            rule = ssg.build_yaml.Rule.from_yaml(rule_path, env_yaml)
+            try:
+                rule = ssg.build_yaml.Rule.from_yaml(rule_path, env_yaml)
+            except ssg.build_yaml.DocumentationNotComplete:
+                # Happens on non-debug build when a rule is "documentation-incomplete"
+                continue
```
Comment (yuumasato) [inline tests/test_parse_affected.py]: For posterity, while merging `stabilization-v0.1.47` into `master`, there was a conflict here.
They come from #4945 and #4962.

**Pull request merged**